client/trino-jdbc/src/main/java/io/trino/jdbc/AbstractTrinoResultSet.java

```diff
@@ -260,7 +260,13 @@ public boolean getBoolean(int columnIndex)
             throws SQLException
     {
         Object value = column(columnIndex);
-        return (value != null) ? (Boolean) value : false;
+        if (value == null) {
+            return false;
+        }
+        if (value instanceof Boolean) {
+            return (Boolean) value;
+        }
+        throw new SQLException("Value is not a boolean: " + value);
```

Comment:
Can we add a test? Probably in `BaseTestJdbcResultSet`.

Reply:
@ebyhr I had add asimple test. I think it's enough.